android/app/src/main/java/com/robertodoering/harpy/AndroidPathProviderPlugin.kt

```diff
@@ -1,20 +1,16 @@
 package com.robertodoering.harpy
 
 import android.os.Environment
+import io.flutter.embedding.engine.plugins.FlutterPlugin
 import io.flutter.plugin.common.MethodCall
-import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
-import io.flutter.plugin.common.PluginRegistry.Registrar
 
-class AndroidPathProviderPlugin : MethodCallHandler {
-  companion object {
-    @JvmStatic
-    fun registerWith(registrar: Registrar) {
-      val channel = MethodChannel(registrar.messenger(), "android_path_provider")
-      channel.setMethodCallHandler(AndroidPathProviderPlugin())
```

Comment:
You mean here ? I just believed it was only for compatibility of old version.

Reply:
There is a difference when you write platform code to an existing flutter app and platform code for a flutter plugin.

If you want to write platform code in harpy, you will have to listen to the `MethodChannel` that you are calling from your dart code (in our case `android_path_provider`) in the `MainActivity.java` (android/app/src/main/java/com/robertodoering/harpy/MainActivity.java)
This step is described [here](https://flutter.dev/docs/development/platform-integration/platform-channels?tab=android-channel-java-tab#step-3-add-an-android-platform-specific-implementation).

The code that you deleted here is used by flutter to generate the plugin registrants when you compile the app.

---

Instead of trying to add the plugin functionality inside platform specific code in harpy, I suggest you instead fork the plugin and upgrade it to use the new plugins api. This way other people can use your updated fork and we don't add this functionality directly into harpy (which should be our last resort).

We can then use your fork as a git dependency in the pubspec.yaml and if the original author has is interested in merging your PR we can eventually switch back to the pub version.

I would be much happier with that

Reply:
Gotcha, appreciate for your patient.